tests/control/DirectorTest.php

```diff
@@ -97,36 +99,41 @@ public function testAbsoluteURL() {
 		$_SERVER['REQUEST_URI'] = "$rootURL/mysite/sub-page/";
 		Config::inst()->update('Director', 'alternate_base_url', '/mysite/');
 
-		//test empty URL
-		$this->assertEquals("$rootURL/mysite/sub-page/", Director::absoluteURL(''));
+		//test empty / local urls
+		foreach(array('', './', '.') as $url) {
+			$this->assertEquals("$rootURL/mysite/", Director::absoluteURL($url, Director::BASE));
+			$this->assertEquals("$rootURL/", Director::absoluteURL($url, Director::ROOT));
+			$this->assertEquals("$rootURL/mysite/sub-page/", Director::absoluteURL($url, Director::REQUEST));
+		}
 
-		//test absolute - /
+		// Test site root url
 		$this->assertEquals("$rootURL/", Director::absoluteURL('/'));
 
-		//test relative
-		$this->assertEquals("$rootURL/mysite/sub-page/", Director::absoluteURL('./'));
-		$this->assertEquals("$rootURL/mysite/sub-page/", Director::absoluteURL('.'));
+		// Test non-relative urls
+		foreach(array(Director::BASE, Director::ROOT, Director::REQUEST) as $method) {
```

Comment:
Small stylistic niggle: Avoid loops in assertions, or ensure all input params (`$method`) are mentioned in the assertion message. Hard to debug otherwise